snippets-app/src/main.rs

```
@@ -1,137 +1,55 @@
-    }
-    let text = fs::read_to_string(STORE_FILE)?;
-    if text.trim().is_empty() {
-        return Ok(Vec::new());
-    }
-    let v: serde_json::Value = serde_json::from_str(&text)
-        .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, format!("JSON parse error: {e}")))?;
-    let arr = v.as_array().cloned().unwrap_or_default();
-    let mut out = Vec::new();
-    for item in arr {
-        let name = item.get("name").and_then(|x| x.as_str()).unwrap_or("").to_string();
-        let lang = item.get("lang").and_then(|x| x.as_str()).unwrap_or("").to_string();
-        let code = item.get("code").and_then(|x| x.as_str()).unwrap_or("").to_string();
-        if !name.is_empty() {
-            out.push(Snippet { name, lang, code });
-        }
-    }
-    Ok(out)
-}
-
-fn save_store(items: &[Snippet]) -> io::Result<()> {
-    let arr: Vec<serde_json::Value> = items
-        .iter()
-        .map(|s| {
-            serde_json::json!({
-                "name": s.name,
-                "lang": s.lang,
-                "code": s.code
-            })
-        })
-        .collect();
-    let text = serde_json::to_string_pretty(&arr)
-        .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e.to_string()))?;
-    fs::write(STORE_FILE, text)
+#[derive(Clone, Serialize, Deserialize, Debug, Default)]
+struct Store {
+    snippets: Vec<Snippet>,
 }
 
-fn print_usage() {
-    eprintln!("usage:");
-    eprintln!("  snippets-app add \"<name>\" \"<lang>\" \"<code>\"");
-    eprintln!("  snippets-app list");
-    eprintln!("  snippets-app show \"<name>\"");
-    eprintln!("  snippets-app delete \"<name>\"");
-}
+impl Store {
+    fn load_store() -> std::io::Result<Self> {
+        if !Path::new(STORE_FILE).exists() {
+            return Ok(Self::default());
+        }
+        let file = File::open(STORE_FILE)?;
+        let reader = BufReader::new(file);
+        let snippets: Vec<Snippet> = serde_json::from_reader(reader).unwrap_or_default();
```

I believe you can return an error here:

```rust
let snippets: Vec<Snippet> = serde_json::from_reader(reader)?;
```